### ISSUE TITLE
Create an elixir supervisor tree for running processes

### ISSUE THREAD

**Issue by alexanderson1993** [Thorium-Sim/thorium#11]
Example processes:
- Engine Heat
- Sensor contact animation
- Any kind of animation

**Comment by alexanderson1993** [comment]
The tree would consist of a flight object at the top, which, based on initialization values, would spawn all of the necessary objects for that flight. The init values would effectively be a snapshot of the flight at any given time.

The init values of a flight would be stored in the database and would allow for quick and easy saving and loading of flights at any given time, starting a flight with specific parameters, etc. It basically initializes state.

The flight object initialization would spawn the simulators, and other global objects (such as sensor contacts). Simulators would spawn stations and other simulator-global objects (like decks, rooms, and personnel). Stations would spawn cards, cards would spawn/update systems. 

In other words, you could serialize the entire state of a flight with this system and then stick it in the database to be loaded at a future time.

This also means that during flights, state is managed entirely on the server and not in the database at all. The database only stores persisted information. Potential persisted information includes saved flights,  the universe of sensor objects, among other things.

**Comment by alexanderson1993** [comment]
Cards, when instantiated, know which systems they should have access to. If the simulator doesn't have those systems, the card should freak out somehow.
